src/gui/src/dbDescriptors.cpp

```diff
@@ -230,6 +230,206 @@ static odb::dbTechLayer* getLayerSelection(odb::dbTech* tech,
 
 ////////
 
+DbTechDescriptor::DbTechDescriptor(odb::dbDatabase* db) : db_(db)
+{
+}
+
+std::string DbTechDescriptor::getName(std::any object) const
+{
+  return "Default";
+}
+
+std::string DbTechDescriptor::getTypeName() const
+{
+  return "Tech";
+}
+
+bool DbTechDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void DbTechDescriptor::highlight(std::any object, Painter& painter) const
+{
+}
+
+Descriptor::Properties DbTechDescriptor::getProperties(std::any object) const
+{
+  auto gui = Gui::get();
+  auto tech = std::any_cast<odb::dbTech*>(object);
+
+  Properties props({{"DbUnits per Micron", tech->getDbUnitsPerMicron()},
+                    {"LEF Units", tech->getLefUnits()},
+                    {"LEF Version", tech->getLefVersionStr()}});
+
+  SelectionSet tech_layers;
+  for (auto tech_layer : tech->getLayers()) {
+    tech_layers.insert(gui->makeSelected(tech_layer));
+  }
+  props.push_back({"Tech Layers", tech_layers});
+
+  SelectionSet tech_vias;
+  for (auto tech_via : tech->getVias()) {
+    tech_vias.insert(gui->makeSelected(tech_via));
+  }
+  props.push_back({"Tech Vias", tech_vias});
```

Comment:
there are also getViaRules() and getViaGenerateRules() that should be included.

Reply:
There is not a descriptor for via-rules, what should it display in the tech inspection?